src/vehicle_autodrive.cpp

```diff
@@ -0,0 +1,1305 @@
+            extent.first = std::min( extent.first, pos.x );
+            extent.second = std::max( extent.second, pos.x );
+        }
+    }
+    for( const auto &extent : extent_map ) {
+        const int y = extent.first;
+        const int x_min = extent.second.first;
+        const int x_max = extent.second.second;
+        for( int x = x_min; x <= x_max; x++ ) {
+            ret.occupied_zone.emplace_back( x, y );
+        }
+    }
+    for( int part_num : driven_veh.rotors ) {
+        const vehicle_part &part = driven_veh.part( part_num );
+        const int diameter = part.info().rotor_diameter();
+        const int radius = ( diameter + 1 ) / 2;
+        if( radius > 0 ) {
+            tripoint pos;
+            driven_veh.coord_translate( tdir, pivot, part.mount, pos );
+            for( tripoint pt : points_in_radius( pos, radius ) ) {
+                ret.occupied_zone.emplace_back( pt.xy() );
+            }
+        }
+    }
+    // figure out the maximum amount of displacement that can happen from advancing the
+    // tileray a single step by advancing it a bunch of times and reducing the x and y
+    // components of the displacement to at most 1
+    constexpr int far_advance = 100;
+    tdir.advance( far_advance );
+    const point increment( signum( tdir.dx() ), signum( tdir.dy() ) );
+    // compute the subset of vehicle points that would move to previously unoccupied spots
+    // (a.k.a. "moving first") when the vehicle moves one step; these are the only points
+    // we need to check for collision when the vehicle is moving in this direction
+    const std::unordered_set<point> occupied_set( ret.occupied_zone.begin(), ret.occupied_zone.end() );
+    for( const point &pt : ret.occupied_zone ) {
+        if( occupied_set.find( pt + increment ) == occupied_set.end() ) {
+            ret.collision_points.emplace_back( pt );
+        }
+    }
+    return ret;
+}
+
+
+// Return true if the map tile at the given position (in map coordinates)
+// can be driven on (not an obstacle).
+bool vehicle::autodrive_controller::check_drivable( tripoint pt ) const
+{
+    const map &here = get_map();
+
+    // check if another vehicle is there; tiles occupied by the current
+    // vehicle are evidently drivable
+    const optional_vpart_position ovp = here.veh_at( pt );
+    if( ovp ) {
+        // Known corner case: some furniture can be driven over, but will collide with
+        // wheel parts; if the vehicle starts over such a furniture we'll mark that tile
+        // as safe and may collide with it by turning; however if we mark it unsafe
+        // we'll have no viable paths away from the starting point.
+        return &ovp->vehicle() == &driven_veh;
+    }
+
+    const tripoint_abs_ms pt_abs( here.getabs( pt ) );
+    const tripoint_abs_omt pt_omt = project_to<coords::omt>( pt_abs );
+    // only check visibility for the current OMT, we'll check other OMTs when
+    // we reach them
+    if( pt_omt == data.current_omt ) {
```

Comment:
> What happens at the junctures between OMTs?

Before crossing into a new OMT: nothing special.
After crossing into a new OMT: the path is recomputed, using only visible or memorized tiles in the new OMT.

Reply:
I'm wondering about if the pivot point (or wherever it's measured from) is not yet in a new OMT, but the front of the vehicle is.

Reply:
Yeah, so that part may be considered cheating a bit because the pathfinding algorithm may choose a path that puts the front of your car through tiles from the next OMT that you can't initially see but it knows that they're safe. This is mitigated by the per-turn visibility and collision checks. The idea is that in practice by the time you get there you will be able to see those tiles; but if it turns out that you can't then autodrive will slow down or stop.